src/Microsoft.VisualStudio.Services.Agent/HostContext.cs

```diff
@@ -175,6 +169,65 @@ public HostContext(HostType hostType, string logFile = null)
             }
         }
 
+        private ILoggedSecretMasker CreateSecretMasker()
+        {
+            // When enabled, use the OSS package-provided secret masker from
+            // Microsoft.Security.Utilities.Core. Otherwise, use the legacy
+            // secret masker from VSO.
+            bool useNewSecretMasker = AgentKnobs.EnableNewSecretMasker.GetValue(this).AsBoolean();
+
+            // When enabled, use add additional regex patterns to the secret
+            // masker. For the OSS package-provided secret masker, this will use
+            // its 'PreciselyClassifiedSecurityKeys'. For the legacy secret
+            // masker, this will use 'CredScanPatterns' implemented in this
+            // repository.
+            bool useAdditionalMaskingRegexes = AgentKnobs.EnableAdditionalMaskingRegexes.GetValue(this).AsBoolean();
+
+#pragma warning disable CA2000 // Dispose objects before losing scope. False positive: LoggedSecretMasker takes ownership.
+            ISecretMasker rawSecretMasker;
+            if (useNewSecretMasker)
+            {
+                IEnumerable<RegexPattern> patterns = Array.Empty<RegexPattern>();
+ 
+                if (useAdditionalMaskingRegexes)
+                {
+                    patterns = WellKnownRegexPatterns.PreciselyClassifiedSecurityKeys;
+                }
+
+                rawSecretMasker = new OssSecretMasker(patterns);
+            }
+            else
+            {
+                rawSecretMasker = new SecretMasker();
+            }
+#pragma warning restore CA2000 // Dispose objects before losing scope
+
+            ILoggedSecretMasker secretMasker = new LoggedSecretMasker(rawSecretMasker);
+
+            secretMasker.AddValueEncoder(ValueEncoders.JsonStringEscape, $"HostContext_{WellKnownSecretAliases.JsonStringEscape}");
+            secretMasker.AddValueEncoder(ValueEncoders.UriDataEscape, $"HostContext_{WellKnownSecretAliases.UriDataEscape}");
+            secretMasker.AddValueEncoder(ValueEncoders.BackslashEscape, $"HostContext_{WellKnownSecretAliases.UriDataEscape}");
+
+            if (useNewSecretMasker)
+            {
+                secretMasker.AddRegex(AdditionalMaskingRegexes.UrlSecretPatternNonBacktracking, $"HostContext_{WellKnownSecretAliases.UrlSecretPattern}");
```

Comment:
>UrlSecretPatternNonBacktracking [](http://example.com/codeflow?start=63&length=31)

My suggestion is to add this pattern using the `RegexPattern` class above, where we have OSSMasker as the `rawSecretMasker`.  The id s/be `SEC101/127`. The name s/be `UrlCredentials`. `DetectionMetadata.MediumConfidence` (or start with Low to be safe, what do you think?

Reply:
Done.